lms/djangoapps/mobile_api/tests.py

```diff
@@ -21,35 +26,233 @@
+
+
+class MobileUserCourseAPITestCase(MobileAPITestCase):
+    """
+    Base class for testing Mobile APIs related to both users and courses.
+    """
+    def get_url(self, reverse_args=None, **kwargs):
+        reverse_args = reverse_args or {}
+        reverse_args.update({
+            'course_id': unicode(kwargs.get('course_id', self.course.id)),
+            'username': kwargs.get('username', self.user.username)
+        })
+        return super(MobileUserCourseAPITestCase, self).get_url(reverse_args, **kwargs)
+
+
+class MobileAuthTestMixin(object):
+    """
+    Test Mixin for testing APIs decorated with MobileView or mobile_view.
+    """
+    def test_no_auth(self):
+        self.logout()
+        self.get_response(expected_response_code=401)
+
+
+class MobileAuthUserTestMixin(MobileAuthTestMixin):
+    """
+    Test Mixin for testing APIs related to users: mobile_view or MobileView with is_user=True.
+    """
+    def test_invalid_user(self):
+        self.login_and_enroll()
+        self.get_response(expected_response_code=403, username='no_user')
+
+    def test_other_user(self):
+        # login and enroll as the test user
+        self.login_and_enroll()
+        self.logout()
+
+        # login and enroll as another user
+        other = UserFactory.create()
+        self.client.login(username=other.username, password='test')
+        self.enroll()
+        self.logout()
+
+        # now login and call the API as the test user
+        self.login()
+        self.get_response(expected_response_code=403, username=other.username)
+
+
+@ddt.ddt
+class MobileCourseAccessTestMixin(object):
+    """
+    Test Mixin for testing APIs marked with mobile_course_access.
+    (Use MobileEnrolledCourseAccessTestMixin when verify_enrolled is set to True.)
+    Subclasses are expected to inherit from MobileAPITestCase.
+    Subclasses can override verify_success, verify_failure, and init_course_access methods.
+    """
+    def verify_success(self, response):
+        """Base implementation of verifying a successful response."""
+        self.assertEqual(response.status_code, 200)
+
+    def verify_failure(self, response):
+        """Base implementation of verifying a failed response."""
+        self.assertEqual(response.status_code, 404)
+
+    def init_course_access(self, course_id=None):
+        """Base implementation of initializing the user for each test."""
+        self.login_and_enroll(course_id)
+
+    def test_success(self):
+        self.init_course_access()
+
+        response = self.get_response(expected_response_code=None)
+        self.verify_success(response)
+
+    def test_course_not_found(self):
+        non_existent_course_id = CourseKey.from_string('a/b/c')
+        self.init_course_access(course_id=non_existent_course_id)
+
+        response = self.get_response(expected_response_code=None, course_id=non_existent_course_id)
+        self.verify_failure(response)
```

Comment:
Calling `self.verify_failure` allows subclasses to handle the failure case.  For example, `TestUserEnrollmentApi` overrides `verify_failure` since it always returns a `200`, but with `0` courses.